registry/client.go

```diff
@@ -488,3 +496,16 @@ func (c *client) GetAgent(agentID string) (*Agent, error) {
 		Owner:    agt.Owner.Hex(),
 	}, nil
 }
+
+func (c *client) RegisterScanner(privateKey *ecdsa.PrivateKey, ownerAddress string, chainID int64, metadata string) (txHash string, err error) {
```

Comment:
can we make the client have the private key as part of its config?  (optional property)

Reply:
i guess maybe it means "RegisterSelfAsScanner"

Reply:
Just made a change. It makes sense but the "self" naming doesn't so I kept it.